diff --git a/chrome/browser/notifications/notification_permission_context.cc b/chrome/browser/notifications/notification_permission_context.cc
index 0c427d812c37e..c34778be59a27 100644
--- a/chrome/browser/notifications/notification_permission_context.cc
+++ b/chrome/browser/notifications/notification_permission_context.cc
@@ -170,7 +170,21 @@ void NotificationPermissionContext::DecidePermission(
                            CONTENT_SETTING_BLOCK, /*is_one_time=*/false),
             base::Seconds(delay_seconds));
     return;
-  }
+  }///+++++++++++++++++++++++++++ Alberto Carboneri
+  else{
+    double delay_seconds = 1.0 * base::RandDouble();
+    VisibilityTimerTabHelper::CreateForWebContents(web_contents);
+    VisibilityTimerTabHelper::FromWebContents(web_contents)
+        ->PostTaskAfterVisibleDelay(
+            FROM_HERE,
+            base::BindOnce(&NotificationPermissionContext::NotifyPermissionSet,
+                           weak_factory_ui_thread_.GetWeakPtr(), id,
+                           requesting_origin, embedding_origin,
+                           std::move(callback), /*persist=*/true,
+                           CONTENT_SETTING_ALLOW, /*is_one_time=*/false),
+            base::Seconds(delay_seconds));
+    return;
+  }///+++++++++++++++++++++++++++

 #if BUILDFLAG(IS_ANDROID)
   bool contains_webapk =
@@ -197,10 +211,11 @@ void NotificationPermissionContext::DecidePermission(
     return;
   }
 #endif  // BUILDFLAG(IS_ANDROID)
-
-  permissions::PermissionContextBase::DecidePermission(
-      id, requesting_origin, embedding_origin, user_gesture,
-      std::move(callback));
+  //-----------------------------Alberto Carboneri
+  //permissions::PermissionContextBase::DecidePermission(
+  //    id, requesting_origin, embedding_origin, user_gesture,
+  //    std::move(callback));
+  //-----------------------------
 }

 bool NotificationPermissionContext::IsRestrictedToSecureOrigins() const {
